DataFormats/Luminosity/interface/PixelClusterCounts.h

```diff
@@ -0,0 +1,56 @@
+#ifndef DataFormats_Luminosity_PixelClusterCounts_h
+#define DataFormats_Luminosity_PixelClusterCounts_h
+/** \class reco::PixelClusterCounts
+ *  
+ * Reconstructed PixelClusterCounts object that will contain the moduleID, BX, and counts.
+ *
+ * \authors: Sam Higginbotham shiggib@cern.ch and Chris Palmer capalmer@cern.ch
+ * 
+ *
+ */
+#include <algorithm>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <vector>
+
+namespace reco {
+class PixelClusterCounts {
+static constexpr unsigned int nEmBX=3564;//Empty BX to fill with counts
+    public:
+        PixelClusterCounts() : m_events(nEmBX){}
+        ////////////////////////////////////////////
+        void increment(int mD,int BXid,int count){
+            std::vector<int>::iterator it;
+            it = std::find(m_ModID.begin(), m_ModID.end(), mD);
+            size_t modIndex = it - m_ModID.begin();
```

Comment:
```c++
 size_t modIndex = std::distance(m_ModID.begin(), std::find(m_ModID.begin(), m_ModID.end(), mD));
 if (modIndex == m_ModID.size()){
     m_ModID.push_back(mD);
     m_counts.resize(m_counts.size()+nEmBX, 0);
}
```

Please add a constructor with a number of modules to be reserved (is it 2528 for the phase-1 total? ) so that we do not have massive reallocations going on.
I suppose that during a run this m_counts vector eventually grows to about 36MB in memory.
Is it correct?